### ISSUE TITLE
lstmLayer_bp: not taking into account gradient contribution from last cell state

### ISSUE THREAD

**Issue by AlexDBlack** [deeplearning4j/deeplearning4j#8838]
Suppose during LSTM forward pass we return the full time series, yLast and cLast.

It appears that if we use the cLast output for something that impacts the loss function, we'll be passing gradients backward through this variable too. The lstmLayer_bp implementation doesn't seem to take it into account.

Java test case: compare line 57 vs. 58
https://gist.github.com/AlexDBlack/41f37a66eff1619e9598588403a25fc0

Line 57 fails gradient check, line 58 passes.

Essentially, the mul(0) guarantees that the dLdcLast passed into the backprop op is 0s, as the loss function now doesn't depend (directly) on that variable/output. Put another way: a non-zero gradient contribution for dLdcLast results in a gradient check failure.

The solution should be simple: at the last time step, just add the passed in dLdcLast array to the calculated dLdc at that time step.

**Comment by AlexDBlack** [comment]
Confirmed fixed, thanks
